src/libraries/System.Private.Uri/src/System/UriHelper.cs

```diff
@@ -193,35 +185,15 @@ internal static class UriHelper
             for (; i < stringToEscape.Length && (c = stringToEscape[i]) <= 0x7F && noEscape[c]; i++) ;
             if (i == stringToEscape.Length)
             {
-                if (dest != null)
-                {
-                    EnsureCapacity(dest, destPos, stringToEscape.Length);
-                    stringToEscape.CopyTo(dest.AsSpan(destPos));
-                    destPos += stringToEscape.Length;
-                }
-
-                return dest;
+                dest.Append(stringToEscape);
             }
+            else
+            {
+                dest.Append(stringToEscape.Slice(0, i));
 
-            // Otherwise, create a ValueStringBuilder to store the escaped data into,
-            // append to it all of the noEscape chars we already iterated through, and
-            // escape the rest into the ValueStringBuilder.
-            var vsb = new ValueStringBuilder(stackalloc char[256]);
-            vsb.Append(stringToEscape.Slice(0, i));
-            EscapeStringToBuilder(stringToEscape.Slice(i), ref vsb, noEscape, checkExistingEscaped);
-
-            // Finally update dest with the result.
-            EnsureCapacity(dest, destPos, vsb.Length);
-            vsb.TryCopyTo(dest.AsSpan(destPos), out int charsWritten);
-            destPos += charsWritten;
-            return dest;
+                ReadOnlySpan<bool> noEscapeCopy = MemoryMarshal.CreateReadOnlySpan(ref MemoryMarshal.GetReference(noEscape), UnreservedReservedTable.Length);
 
-            static void EnsureCapacity(char[]? dest, int destSize, int requiredSize)
```

Comment:
This looks like a bug in the current implementation as the param is not `ref char[]`, therefore the change isn't visible.

Reply:
Do we need a test that would catch it?

Reply:
The overload accepting a char[] destination was only called from ReCreateParts, after [this conservative worst-case capacity calculation](https://github.com/dotnet/runtime/blob/c991d9bc35b38016d8a48e745f842b648678fb9a/src/libraries/System.Private.Uri/src/System/Uri.cs#L2711-L2718). I believe it was thankfully unreachable code, as we never needed a resize.